### DISCUSSION TITLE
NXP : i.MXRT1170 : RPmsg  (or any kind of inter-core comm) support

### DISCUSSION THREAD

**Discussion by mtitinger** [zephyrproject-rtos/zephyr#61032] [General] [answered]
Hello community,

while doing a quick browsing of the sources, I saw support for RPmsg, but I'd couldn't see use of the Messaging Unit IP in the case of RT1170. I would appreciate some input regarding inter-core comm with the i.MXRT platform with Zephyr, what would be the way to go in Zephyr currently : generic RPmsg implem, hand-crafted MU scheme ?  Anyone with feedback with this ? 

Thanks and Regards,
Marc.

**Comment by DerekSnell** [comment, marked as answer]
Hi @mtitinger ,
The i.MX RT multi-core MCUs use OpenAMP for the inter-core communication, which uses the RPmsg protocol.  On the i.MX RT1170 and RT1160 families, OpenAMP does leverage the Messaging Unit hardware.  OpenAMP is tested on these platforms using the [OpenAMP sample](https://docs.zephyrproject.org/latest/samples/subsys/ipc/openamp/README.html#openamp-sample-application).  And the MIMXRT1170-EVK board page includes details on [Building a Dual-Core Image](https://docs.zephyrproject.org/latest/boards/arm/mimxrt1170_evk/doc/index.html#building-a-dual-core-image).

I hope that helps, best regards

**Reply by mtitinger** [reply]
Hello @DerekSnell, thanks a lot ! sorry I overlooked it. 

Kind regards,
Marc.
